src/Controls/tests/BindingSourceGen.UnitTests/BindingRepresentationGenTests.cs

```diff
@@ -72,6 +56,165 @@ public void GenerateBindingWithNestedProperties()
                 ],
                 true
             );
+
+        //TODO: Change arrays to custom collections implementing IEquatable
+        Assert.Equal(expectedBinding.Path, actualBinding.Path);
+        Assert.Equivalent(expectedBinding, actualBinding, strict: true);
+    }
+
+    [Fact]
+    public void GenerateBindingWithNullableElementInPath()
+    {
+        var source = """
+        using Microsoft.Maui.Controls;
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (Foo f) => f.Button?.Text.Length);
+
+        class Foo
+        {
+            public Button? Button { get; set; }
+        }
+        """;
+
+        var actualBinding = SourceGenHelpers.GetBinding(source) with { Id = 0 }; // TODO: Improve indexing of bindings
+        var expectedBinding = new CodeWriterBinding(
+                0,
+                new SourceCodeLocation("", 3, 7),
+                new TypeName("global::Foo", false, false),
+                new TypeName("int", true, false),
+                [
+                    new PathPart("f", false),
```

Comment:
Now that I'm thinking about it, I think we should skip the first item in the list that is the identifier. I think that in the code writer part, I just always called it `source` and I assumed that the path starts with the first property/indexer. Can you please check if it is the case? Maybe we should add a simple end-to-end snapshot test for this case.